### PR TITLE
Upgrade rspec to later version

### PR DESCRIPTION
**Pull request by surminus**
the last_comment method was removed in Rake, and this is fixed in a later version of rspec.

Ref: https://github.com/rspec/rspec-core/pull/2197

### CODE REVIEW

**Comment by annashipman** [comment]
👍

**Comment by surminus** [comment]
Gah, didn't fix it 🤔

**Comment by annashipman** [comment]
Well it got past it on the first run and now is struggling with nokogiri... I refer you to bob...

**Comment by surminus** [comment]
I've added a bunch more changes which _should_ fix it for all versions.

**Comment by surminus** [comment]
I've repushed the fix and made commits more specific, and bumped the minor version.

Getting these tests fixed will help fix up the fog dependency which is a bug I've just hit when trying to deploy something 😢 

/cc @annashipman @rjw1

**Comment by annashipman** [comment]
This looks good to me: the tests pass, well done!

I'm not super-keen on pinning Rake to a version less than 11 though - is there a future in which we can use the latest Rake?

**Comment by surminus** [comment]
I'll investigate fixing this with a more up to date Rake.

**Comment by surminus** [comment]
Added several more commits. The error was actually being thrown by rubocop, so we have a newer Rake and an updated rubocop, which also involved a few syntax updates.

**Comment by surminus** [comment]
Failing on Ruby 2.0.0 🤕 

```
Failure/Error:
    SimpleCov.adapters.define 'gem' do
      add_filter '/spec/'
      add_filter '/vendor/'
  
      add_group 'Libraries', '/lib/'
    end
```

**Comment by rjw1** [comment]
is 2.0.0 still a supported ruby?

**Comment by surminus** [comment]
Ruby 2.0.0 is no longer supported. We can remove this support as it breaks a lot of things.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 2.0.1 (2017-05-30)
```
Comment (annashipman) [inline CHANGELOG.md]: Today's date instead?

**Comment by annashipman** [comment]
This all looks good to me and well done for fixing all those errors. Let's remove support for Ruby 2.0.0. next time it causes a problem?

I've got one tiny comment on the date but other than that, 👍 merge away!

**Comment by surminus** [comment]
Fixed the date! One thing to note is that since we moved CI the gem is no longer automatically published to rubygems, so I'll need do the work to ensure further new versions are pushed. I believe this has caused a problem with other dependencies.

**Comment by surminus** [comment]
I manually published the new gem (not sure why I'm doing this on a Sunday):

https://rubygems.org/gems/vcloud-core/versions/2.0.1

**Comment by annashipman** [comment]
You definitely [shouldn't be doing it on a Sunday](http://dan.carley.co/blog/2014/05/21/working-late-responsibly/) but 👍 well done for getting this finished. Thanks!

**Pull request merged**